pmemkv/src/main/java/site/ycsb/db/pmemkv/PmemKVClient.java

```diff
@@ -119,31 +121,57 @@ public void init() throws DBException {
         String engineName = props.getProperty(ENGINE_PROPERTY, "cmap");
 
         String path = props.getProperty(PATH_PROPERTY);
+        String size = props.getProperty(SIZE_PROPERTY);
+        String jsonConfigProp = props.getProperty(JSON_CONFIG_PROPERTY);
+        Path jsonConfigFile = null;
+        String jsonConfigContent = null;
+        if (jsonConfigProp != null && !jsonConfigProp.isEmpty()) {
+          jsonConfigFile = Paths.get(jsonConfigProp);
+          try {
+            if (jsonConfigFile != null) {
+              jsonConfigContent = new String(Files.readAllBytes(jsonConfigFile), UTF_8);
+            }
+          } catch (IOException e) {
+            throw new DBException(JSON_CONFIG_PROPERTY +
+              " contain faulty path or points to improper json config params");
+          }
+        }
+
         if (path == null) {
           throw new DBException(PATH_PROPERTY + " is obligatory to run PmemKV client");
         }
-        String size = props.getProperty(SIZE_PROPERTY);
         if (size == null) {
           throw new DBException(SIZE_PROPERTY + " is obligatory to run PmemKV client");
         }
+
         boolean startError = false;
+        Database.Builder<byte[], Map<String, ByteIterator>> builder = null;
+
+        /* try to open db first */
         try {
-          /* try to open db first */
-          db = new Database.Builder<byte[], Map<String, ByteIterator>>(engineName)
-              .setSize(Long.parseLong(size))
+          builder = new Database.Builder<byte[], Map<String, ByteIterator>>(engineName);
+          if (jsonConfigContent != null) {
+            builder = builder.fromJson(jsonConfigContent);
+          }
+          db = builder
               .setPath(path)
+              .setSize(Long.parseLong(size))
               .setKeyConverter(new ByteConverter())
               .setValueConverter(new MapToByteBufferConverter())
               .build();
         } catch (DatabaseException e) {
           startError = true;
         }
         if (startError) {
+          /* or create it, if it doesn't exist */
           try {
-            /* or create it, if it doesn't exist */
-            db = new Database.Builder<byte[], Map<String, ByteIterator>>(engineName)
-                .setSize(Long.parseLong(size))
+            builder = new Database.Builder<byte[], Map<String, ByteIterator>>(engineName);
+            if (jsonConfigContent != null) {
+              builder = builder.fromJson(jsonConfigContent);
+            }
+            db = builder
                 .setPath(path)
+                .setSize(Long.parseLong(size))
                 .setKeyConverter(new ByteConverter())
                 .setValueConverter(new MapToByteBufferConverter())
                 .setForceCreate(true)
```

Comment:
Hm, right now, this logic of opening/creating db kind of ignores the json config. If someone would like to open (or error if the db does not exists) the db (by not specificing any flag in the config) we will still try to create the file. Maybe, instead of having this create/open logic here, we could provide a base json config which would be used by default. It could have "create_if_missing":true

Reply:
And then, users could change the behavior by just modyfing/extending/chaning this base config.

Reply:
as I wrote in the comment above, I'd not change pmemkv to 1.5.0 and use just pmemkv-java 1.2.0 for now (btw we don't officially support these new flags in java, yet)

If we introduce it there, we can update YCSB

Reply:
Also, if anyone set a flag in the json file it will be taken into account, since the first "open", does not use any flags...

nonetheless, this solution may not be the best, but it's easy to use for anyone and does not require setting anything other than path and size (which is exposed as a `pmemkv.<param>`)